tutorials/intermediate/arranging_axes.py

```diff
@@ -0,0 +1,360 @@
+# See below for the description of how to do the same thing using
+# `~matplotlib.gridspec.GridSpec` or `~matplotlib.pyplot.subplot2grid`.
+#
+# Variable widths or heights in a grid
+# ------------------------------------
+#
+# Both `~.pyplot.subplots` and `~.pyplot.subplot_mosaic` allow the rows
+# in the grid to be different heights, and the columns to be different
+# widths using the *gridspec_kw* keyword argument.
+# Spacing parameters accepted by :class:`~matplotlib.gridspec.GridSpec`
+# can be passed to `~matplotlib.pyplot.subplots` and
+# `~matplotlib.pyplot.subplot_mosaic`:
+
+gs_kw = dict(width_ratios=[1.4, 1], height_ratios=[1, 2])
+fig, axd = plt.subplot_mosaic([['upleft', 'right'],
+                               ['loleft', 'right']],
+                              gridspec_kw=gs_kw, figsize=(4.5, 3.5),
+                              constrained_layout=True)
+for k in axd.keys():
+    annotate_axes(axd[k], f'axd["{k}"]', fontsize=14)
+fig.suptitle('plt.subplot_mosaic()')
+
+############################################################################
+# Nested Axes layouts
+# -------------------
+#
+# Sometimes it is helpful to have two or more grids of Axes that
+# may not need to be related to one another.  The most simple way to
+# accomplish this is to use `.Figure.subfigures`.  Note that the alignement
+# of the subfigure layouts is independent with the Axes spines in each
+# subfigure having independent positions.  See below for a more verbose
+# way to acheive the same effect with `~.gridspec.GridSpecFromSubplotSpec`.
+
+fig = plt.figure(constrained_layout=True)
+subfigs = fig.subfigures(1, 2, wspace=0.07, width_ratios=[1.5, 1.])
+axs0 = subfigs[0].subplots(2, 2)
+subfigs[0].set_facecolor('0.9')
+subfigs[0].suptitle('subfigs[0]\nLeft side')
+subfigs[0].supxlabel('xlabel for subfigs[0]')
+
+axs1 = subfigs[1].subplots(3, 1)
+subfigs[1].suptitle('subfigs[1]')
+subfigs[1].supylabel('ylabel for subfigs[1]')
+
+############################################################################
+# It is also possible to nest Axes using `~.pyplot.subplot_mosaic` using
+# nested lists.  This method does not use subfigures, like above, so lacks
+# the ability to add per-subfigure ``suptitle`` and ``supxlabel``, etc.
+# Rather it is a conveneince wrapper around the `~.SubplotSpec.subgridspec`
+# method described below.
+
+inner = [['innerA'],
+         ['innerB']]
+outer = [['upleft',  inner],
+          ['lowleft', 'lowright']]
+
+fig, axd = plt.subplot_mosaic(outer, constrained_layout=True)
+for k in axd.keys():
+    annotate_axes(axd[k], f'axd["{k}"]')
+
+############################################################################
+# Low-level and advanced grid methods
+# ===================================
+#
+# `~.pyplot.subplots` and `~.pyplot.subplot_mosaic` provide high-level
+# interfaces to create `~.gridspec.GridSpec` objects and add associated
+# new Axes to a figure at given `~.gridspec.SubplotSpec` inside the
+# *GridSpec*.
```

Comment:
It's a bit confusing that you start with "`~.pyplot.subplots` and `~.pyplot.subplot_mosaic` provide high-level interfaces" when you want to focus on low-level structure. It's not obvious that GridSpec and and SubplotSpec are low-level constructs.

Maybe:

> Internally, the arrangement of a grid of Axes is controlled via `GridSpec` and `SubplotSpec`. `GridSpec` defines a (possibly non-uniform) grid of cells. Indexing into the `GridSpec` returns a `SubplotSpec` that covers one or more grid cells, and can be used to sprecify the location of an subplot.
>
> Note: We call an Axes that is positioned via a GridSpec a *subplot*.
>
> The following examples show how to use low-level methods to arrange Axes according to a GridSpec.

Reply:
Great!  lightly modified to

```
# Internally, the arrangement of a grid of Axes is controlled by creating
# instances of `~.GridSpec` and `~.SubplotSpec`. GridSpec defines a
# (possibly non-uniform) grid of cells. Indexing into the GridSpec returns
# a SubplotSpec that covers one or more grid cells, and can be used to
# specify the location of an Axes or "subplot".
#
# The following examples show how to use low-level methods to arrange Axes
# using GridSpec objects.
```